flake.nix

```diff
@@ -1,29 +1,80 @@
 {
-  description = "Template for Holochain app development";
+  description = "Flake for Android + Holochain app development";
 
   inputs = {
     holonix.url = "github:holochain/holonix?ref=main-0.4";
-
     nixpkgs.follows = "holonix/nixpkgs";
     flake-parts.follows = "holonix/flake-parts";
+    rust-overlay.follows = "holonix/rust-overlay";
 
-    p2p-shipyard.url = "github:darksoil-studio/p2p-shipyard/main-0.4";
+    android-nixpkgs = {
+      url = "github:tadfisher/android-nixpkgs";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
   };
-
-  outputs = inputs:
-    inputs.flake-parts.lib.mkFlake { inherit inputs; } {
+  outputs = inputs@{ flake-parts, nixpkgs, rust-overlay, ... }:
+    flake-parts.lib.mkFlake { inherit inputs; } {
       systems = builtins.attrNames inputs.holonix.devShells;
       perSystem = { inputs', config, pkgs, system, ... }: {
-        devShells.default = pkgs.mkShell {
-          inputsFrom = [ 
-            inputs'.p2p-shipyard.devShells.holochainTauriAndroidDev 
-            inputs'.holonix.devShells.default
-          ];
+        formatter = pkgs.nixpkgs-fmt;
+
+        devShells.default =
+          let
+            overlays = [ (import rust-overlay) ];
+            pkgs = import nixpkgs {
+              inherit system overlays;
+            };
+
+            rust = (pkgs.rust-bin.stable.latest.minimal.override
+              {
+                extensions = [ "clippy" "rustfmt" "rust-src" ];
+                targets = [
+                  "wasm32-unknown-unknown"
+                  "aarch64-linux-android"
+                  "x86_64-linux-android"
+                  "i686-linux-android"
+                  "armv7-linux-androideabi"
+                ];
+              });
+
+            android = inputs.android-nixpkgs.sdk.${system} (sdkPkgs: with sdkPkgs; [
+              cmdline-tools-latest
+              build-tools-34-0-0
+              platform-tools
+              platforms-android-34
+              ndk-25-2-9519653
+            ]);
+          in
+          pkgs.mkShell {
+            packages = [
+              rust
+            ] ++ (with inputs'.holonix.packages; [
+              holochain
+              lair-keystore
+              hc-launch
```

Comment:
Is `hc-launch` actually used here? That just reminded me that we may want to potentially upgrade hc-launch to tauri 2 at some point.

Reply:
Yeah good point. Actually none of those packages are used here. I was imagining consumer apps using this flake, but maybe I should revise it to include with a template for consumer apps.